### ISSUE TITLE
For <p-checkbox> ng-invalid class is getting appended after adding required attribute

### ISSUE THREAD

**Issue by akansha-srivastav-alight** [primefaces/primeng#19408]
### Describe the bug

When we are adding attribute **required="true"** for **p-checkbox ng-invalid class is getting appended** and NVDA is announcing as **checkbox not checked invalid entry**

<img width="1900" height="1140" alt="Image" src="https://github.com/user-attachments/assets/981eb830-fc83-4eee-83a8-a319dad2e36c" />

### Reason for not contributing a PR

- [ ] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/tmav1nst?file=src%2Fapp%2Fcheckbox-filled-demo.ts

### Environment

I am working on an Angular application(v 18) using the version 17 of primeng. The issue is reproducible on both Chrome and Edge. 

### Angular version

18

### PrimeNG version

v21

### Steps to reproduce the behavior

1. Go to https://stackblitz.com/edit/tmav1nst?file=src%2Fapp%2Fcheckbox-filled-demo.ts
2. open NVDA
3. Observe the behaviour

### Expected behavior

Whenever we are adding required=true then the ng-invalid class should not be appended on p-checkbox. NVDA should not announce checkbox unchecked invalid entry

**Comment by kuldeep-kumar-alight** [comment]
Hi @mehmetcetin01140 , could you please look into this issue , we raised a month ago, i have not see any feedback here.

**Comment by kuldeep-kumar-alight** [comment]
@cetincakiroglu for your reference.

**Comment by kuldeep-kumar-alight** [comment]
@cetincakiroglu  @mehmetcetin01140  screen reader NVDA start announcing wrong info as invalid entry, when we are passing required="true" in p-checkbox,  but without required property it is announcing correctly( NVDA is not announcing invalid entry).so as per the accessibility standard check should only announce "name checkbox check required" not "name checkbox check required invalid entry". could you please look into this issue. Thanks
